src/test/scala/com/fulcrumgenomics/bam/api/SamRecordTest.scala

```diff
@@ -289,4 +289,48 @@ class SamRecordTest extends UnitSpec with OptionValues {
     rec1.matesOverlap shouldBe None // Mate's start is not enclosed by rec, and mate's end cannot be determined
     rec2.matesOverlap.value shouldBe true // Mate's start is enclosed by rec, regardless of where mate end is
   }
+
+  "SamRecord.readPosAtRefPos" should "return the correct value" in {
+    case class ReadPosAtRefPos(cigar: String, refPos: Int, readPos: Option[Int], returnLastBaseIfDeleted: Boolean)
+    val testCases = Seq(
```

Comment:
Here and below I think a lot more test cases are needed.  For soft-clipping I think you need tests where:
1. you have leading soft-clipping and request the read position at `rec.start` and `rec.start + 1`
2. you have trailing soft-clipping and you request `read.end` and `read.end - 1`

I think there should also be cases here where there is hard-clipping (should just be ignored).  And skips (e.g. `100N`) in the cigar (presumably return None.